### PR TITLE
Refactor scenario options window

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Some minor refactor work, primarily focussed on removing globals from the scenario options window.

* Use GameState in ScenarioOptions window
* Replace _commonFormatArgs with FormatArguments in scenario options
* Remove references to clickRepeatTicks outside of MouseInput.cpp

### CODE REVIEW

File: src/OpenLoco/src/Windows/ScenarioOptions.cpp [hunk 1]
```diff
@@ -469,7 +462,8 @@ namespace OpenLoco::Ui::Windows::ScenarioOptions
                 widgets[widx::time_limit_value].type = WidgetType::textbox;
                 widgets[widx::time_limit_value_down].type = WidgetType::button;
                 widgets[widx::time_limit_value_up].type = WidgetType::button;
-                _commonFormatArgs[3] = Scenario::getObjective().timeLimitYears;
+                args.skip(2);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/ScenarioOptions.cpp:465]: I don't think this is correct. It needs to be relative to the above pushes that varied in size from 2 and 1 slots. This was to go in the 3 slot regardless. I think just add a skip 1 on L436 .

**Review by duncanspumpkin — APPROVED** [review]
One small issue other than that looks good.

File: src/OpenLoco/src/Windows/ScenarioOptions.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/Windows/ScenarioOptions.cpp:465]: You're right this skip isn't enough in the case of `ObjectiveType::performanceIndex`. Thinking about this, though, I think it's more complicated. `_commonFormatArgs` was defined as an uint16_t array in this file, while FormatArguments uses std::bytes internally. So position 3 here will be equivalent to position 6 in the new situation.

So I'm inclined to leave this skip(2) here, while adding another skip(2) on line 436. Thoughts?

Comment (duncanspumpkin) [reply src/OpenLoco/src/Windows/ScenarioOptions.cpp:465]: Yes you are quite right should be two skip(2)'s I thought skip was in words as well.

File: src/OpenLoco/src/Windows/ScenarioOptions.cpp [hunk 2]
```diff
@@ -426,25 +417,27 @@ namespace OpenLoco::Ui::Windows::ScenarioOptions
...
                 case Scenario::ObjectiveType::vehicleProfit:
-                    *(int32_t*)&*_commonFormatArgs = Scenario::getObjective().monthlyVehicleProfit;
+                    args.push<int32_t>(Scenario::getObjective().monthlyVehicleProfit);
                     widgets[widx::objective_value].text = StringIds::challenge_monetary_value;
                     break;
 
                 case Scenario::ObjectiveType::performanceIndex:
-                    *(int16_t*)&*_commonFormatArgs = Scenario::getObjective().performanceIndex * 10;
+                    args.push<int16_t>(Scenario::getObjective().performanceIndex * 10);
                     widgets[widx::objective_value].text = StringIds::challenge_performance_index;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/ScenarioOptions.cpp:436]:
```suggestion
                    args.skip(2);
                    widgets[widx::objective_value].text = StringIds::challenge_performance_index;
```

**Pull request merged**